text/0038-output-types-for-tokenization-and-sc/0038-output-types-for-tokenization-and-sc.md

```diff
@@ -0,0 +1,3416 @@
+  address.
+
+<details>
+<summary>Return Amount Block</summary>
+<blockquote>
+        Defines the amount of IOTAs that have to be returned to <i>Sender</i>.
+</blockquote>
+</details>
+
+<table>
+    <tr>
+        <td><b>Name</b></td>
+        <td><b>Type</b></td>
+        <td><b>Description</b></td>
+    </tr>
+    <tr>
+        <td>Block Type</td>
+        <td>uint8</td>
+        <td>
+            Set to <strong>value 2</strong> to denote a <i>Return Amount Block</i>.
+        </td>
+    </tr>
+    <tr>
+        <td>Return Amount</td>
+        <td>uint64</td>
+        <td>
+            Amount of IOTA coins the consuming transaction should deposit to the address defined in <i>Sender Block</i>.
+        </td>
+    </tr>
+</table>
+
+This feature block makes it possible to send small amounts of IOTA coins or native tokens to addresses without having
+to lose control of the required dust deposit. It is also a vehicle to send on-chain requests to ISCP chains that do not
+require fees. To prevent the receiving party from blocking access to the dust deposit, it is advised to be used
+together with the [Expiration Blocks](#Expiration-Blocks). The receiving party then has a sender-defined time window to
+agree to the transfer by consuming the output, or the sender regains total control after expiration.
+
+#### Timelock Blocks
+
+The notion of time in the Tangle is introduced via milestones. Each milestone
+[carries the current milestone index and the unix timestamp](https://github.com/jakubcech/protocol-rfcs/blob/jakubcech-milestonepayload/text/0019-milestone-payload/0019-milestone-payload.md#structure) corresponding to that index. Whenever a new milestone appears, nodes perform the white-flag ordering and transaction validation on its past cone. The timestamp and milestone index of the confirming milestone provide the time as an input parameter to transaction validation.
+
+An output that contains a <i>Timelock Block</i> can not be unlocked before the specified timelock has expired. The
+timelock is expired when the timestamp and/or milestone index of the confirming milestone is past the timestamp and/or
+milestone defined in the <i>Timelock Block</i>.
+
+The timelock can be specified as a unix timestamp or as a milestone index. When specified in both ways, both conditions
+have to pass in order for the unlock to be valid.
+
+The two time representations help to protect against the possible downtime of the Coordinator. If the Coordinator is
+down, "milestone index clock" essentially stops advancing, while "real time clock" does not. An output that specifies
+time in both clocks must satisfy both conditions (AND relation).
+
+##### Additional semantic transaction validation rules:
+- An output that has <i>Timelock Milestone Index Block</i> specified must only be consumed and unlocked in a
+  transaction, if the confirming milestone index is > than the `Milestone Index` specified in the block.
```

Comment:
The table says:
> The milestone index/unix starting from which the output can be consumed.

which would imply it can be consumed at a bigger equal index as the one specified in the block. Same applies to the unix timelock. Clarify whether you can consume the input at the exact same index/ts or whether it must be bigger.

Reply:
For the expiration block we say that starting from the expiration index the sender can consume it. For consistencty, let's use the same logic gor the timelock block. So yes, you should be allowed to consume the output starting from the timelock index. I'll add the greater or equal part here.